advec1d.py

```diff
@@ -0,0 +1,97 @@
+'''
+Copyright (C) 2017-2019  Waterloo Quantitative Consulting Group, Inc.
+See COPYING and LICENSE files at project root for more details.
+'''
+
+import numpy as np
+from lib import pyblitzdg
+import matplotlib.pyplot as plt
+
+def advec1dComputeRHS(u,c, nodes1d):
+    vmapM = nodes1d.vmapM
+    vmapP = nodes1d.vmapP
+    mapI = nodes1d.mapI
+    mapO = nodes1d.mapO
+    nx = nodes1d.nx
+    rx = nodes1d.rx
+    Dr = nodes1d.Dr
+    Lift = nodes1d.Lift
+    Fscale = nodes1d.Fscale
+
+    uC = np.squeeze(np.reshape(u, (u.shape[0]*u.shape[1],1), order='F'), axis=1)
+    nxC = np.squeeze(np.reshape(nx, (nx.shape[0]*nx.shape[1],1), order='F'), axis=1)
+    RHS = 0*vmapM
+    uM = uC[vmapM]
+    uP = uC[vmapP]
+
+    # set bc's (outflow condition, and no inflow).
+    uP[mapO] = uM[mapO]
+    uP[mapI] = 0
+
+    # compute jump in flux with Lax-Friedrichs numerical flux.
+    alpha = 0
+    du = (uM - uP)*0.5*(c*nxC - (1-alpha)*np.abs(c*nxC)); 
+
+    RHS =-c*rx*np.dot(Dr, u)
+    RHS += np.dot(Lift, Fscale*np.reshape(du, nx.shape, order='F'))
+
+    return RHS
+
+# Main solver:
+
+xmin =-1.0
+xmax = 4.0
+c = 0.1
+
+finalTime = 20.0
+t = 0.0
+
+# Numerical parameters:
+N = 1
+K = 60
+CFL = 0.25
+
+nodes1d = pyblitzdg.Nodes1DProvisioner(N, K, xmin, xmax)
+nodes1d.buildNodes()
+nodes1d.computeJacobian()
+x = nodes1d.xGrid
+
+u = np.exp(-10*(x*x))
+xplot = np.squeeze(np.reshape(x, ((N+1)*K, 1)))
```

Comment:
Same as comment at line 22 - seems more complex then it could/should be

Reply:
Will look into getting a vector from a 2D array.

Reply:
looks like we probably want this: https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.flatten.html

Reply:
Yep that's a big improvement